lib/galaxy/app_unittest_utils/toolbox_support.py

```diff
@@ -0,0 +1,193 @@
+            config_files,
+            tool_root_dir,
+            app,
+        )
+        # Need to start thread now for new reload callback to take effect
+        self.app.watchers.start()
+
+
+class BaseToolBoxTestCase(TestCase, UsesTools):
+    _toolbox: Optional[SimplifiedToolBox] = None
+
+    @property
+    def integrated_tool_panel_path(self):
+        return os.path.join(self.test_directory, "integrated_tool_panel.xml")
+
+    def assert_integerated_tool_panel(self, exists=True):
+        does_exist = os.path.exists(self.integrated_tool_panel_path)
+        if exists:
+            assert does_exist
+        else:
+            assert not does_exist
+
+    @property
+    def toolbox(self):
+        if self._toolbox is None:
+            self.app._toolbox = self._toolbox = SimplifiedToolBox(self)
+        return self._toolbox
+
+    def setUp(self):
+        self.reindexed = False
+        self.setup_app()
+        install_model = mapping.init("sqlite:///:memory:", create_tables=True)
+        self.app.tool_cache = ToolCache()
+        self.app.install_model = install_model
+        self.app.reindex_tool_search = self.__reindex  # type: ignore[assignment]
+        itp_config = os.path.join(self.test_directory, "integrated_tool_panel.xml")
+        self.app.config.integrated_tool_panel_config = itp_config
+        self.app.watchers = ConfigWatchers(self.app)
+        self._toolbox = None
+        self.config_files = []
+
+    def tearDown(self):
+        self.app.watchers.shutdown()
+
+    def _repo_install(self, changeset, config_filename=None):
+        metadata = {
+            "tools": [
+                {
+                    "add_to_tool_panel": False,  # to have repository.includes_tools_for_display_in_tool_panel=False in InstalledRepositoryManager.activate_repository()
+                    "guid": f"github.com/galaxyproject/example/test_tool/0.{changeset}",
+                    "tool_config": "tool.xml",
+                }
+            ],
+        }
+        if config_filename:
+            metadata["shed_config_filename"] = config_filename
+        repository = tool_shed_install.ToolShedRepository(metadata_=metadata)
+        repository.tool_shed = DEFAULT_TEST_REPO.tool_shed
+        repository.owner = DEFAULT_TEST_REPO.owner
+        repository.name = DEFAULT_TEST_REPO.name
+        repository.changeset_revision = changeset
+        repository.installed_changeset_revision = changeset
+        repository.deleted = False
+        repository.uninstalled = False
+        self.app.install_model.context.add(repository)
+        session = self.app.install_model.context
+        with transaction(session):
+            session.commit()
+        return repository
+
+    def _setup_two_versions(self):
+        self._repo_install(changeset="1")
+        version1 = tool_shed_install.ToolVersion()
+        version1.tool_id = "github.com/galaxyproject/example/test_tool/0.1"
+        self.app.install_model.context.add(version1)
```

Comment:
I think the `context` for [backward compatibility](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/model/base.py#L67)? Given that `context` here *is* the session, do you want to switch to calling it `session` (or sa_session, etc.) for consistency? (i.e., `self.app.install_model.session`)

Reply:
I think that comment meant context.current was the backward compatible thing. I'm fine with install_model.session but it hasn't been my go to for tens of thousands of lines of code since that comment 😅. We should document a best practice here maybe.